hw/ip/otbn/data/otbn.hjson

```diff
@@ -197,6 +197,25 @@
         }
       ],
     }
+    { name: "CTRL",
```

Comment:
just to walk through the thinking.. since this can only be changed while IDLE, we probably don't need an explicit software regwen right? Are there any errors that OTBN might produce outside of a transaction that might make sense to get promoted here?

Reply:
> we probably don't need an explicit software regwen right?

Looking at kmac as an example perhaps I should have a regwen here. The kmac `CFG` register has a regwen but it's software read-only, with the write enabled being controlled by hardware. I should keep things consistent and do the same here.

> Are there any errors that OTBN might produce outside of a transaction that might make sense to get promoted here?

All such errors will be fatal anyway. The definition of 'recoverable error' states 

> Recoverable errors can only occur during the execution of software on OTBN

Reply:
> Looking at kmac as an example perhaps I should have a regwen here. The kmac CFG register has a regwen but it's software read-only, with the write enabled being controlled by hardware. I should keep things consistent and do the same here.

I'm not sure whether that's really needed. Presumably, it would only be used to report the current status ("can I write this?") to SW. But SW can already get that information from the STATUS register.

Reply:
yeah i'm not entirely sure you need regwen either, especially if it cannot be modified during a transaction, and there are no outside-of-transaction recoverable errors.

On that last point though, what do you guys do with storage-update-errors? Because storage-update errors are outside of transaction but not technically a fatal fault (could be a software error).  On the other blocks I'm looking at, I'm starting to have concepts of transactional and non-transactional errors.  This might not apply to you since you don't have shadow registers right now, but it's worth thinking about.

Reply:
Ok I won't add the regwen, just leave things as is.

With regards to storage-update-error can you give an example of what you mean? Any integrity error on a TL transaction results in a fatal error (causing a fatal alert) and any attempt to write dmem/imem from Ibex whilst OTBN is running also causes a fatal error.

Reply:
Storage-update-error is an inherent concept of the shadow reg prim - when the two copies/halves of the shadow are not in sync (presumably because of an attempted fault/glitch attack).  As @tjaychen noted you don't have any shadow regs right now but these errors could occur outside of the scope of a transaction.

Reply:
yeah if you look at @vogelpi aes module, you'll see the shadow registers. Shadow registers have to be written in a specific way.  If we don't follow that method, it results in an "update error".